presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -432,8 +440,12 @@ public static long diffDate(ConnectorSession session, @SqlType("varchar(x)") Sli
     @SqlType(StandardTypes.BIGINT)
     public static long diffTime(ConnectorSession session, @SqlType("varchar(x)") Slice unit, @SqlType(StandardTypes.TIME) long time1, @SqlType(StandardTypes.TIME) long time2)
     {
-        ISOChronology chronology = getChronology(session.getTimeZoneKey());
-        return getTimeField(chronology, unit).getDifferenceAsLong(time2, time1);
+        if (session.isLegacyTimestamp()) {
+            ISOChronology chronology = getChronology(session.getTimeZoneKey());
+            return getTimeField(chronology, unit).getDifferenceAsLong(time2, time1);
+        }
+
+        return getTimeField(UTC_CHRONOLOGY, unit).getDifferenceAsLong(time2, time1);
```

Comment:
Hm... but some zones had rule changes in 1970-01-01. 
I think _generally_ legacy TIME is dealt with session zone, and new TIME is dealt with UTC zone. Doing anything different would be confusing and requiring explanation (like we do eg when we extract seconds from TIME / TIMESTAMP)

Reply:
I did NOT request changing this code in anyway.

I was asking for a comment to acknowledge what a reader might be thinking when reading this code. (i.e. the then branch and else branch is doing the same thing)

But I acknowledge that this is the right thing to do. And I don't feel strongly about adding such comments.

I'll leave this to your discretion.

Reply:
I see, I will add a comment. 

> BTW, I'm curious as to what zone has rule change on 1970-01-01.

America/Bahia_Banderas, America/Hermosillo and few others, all in Mexico (but they are not effectively one zone)

Reply:
oh, i already added the comment: `// Session zone could have policy change on/around 1970-01-01, so we cannot use UTC`

> One additional reason I'm trying to have a comment here: there should not be branches on legacy timestamp flag here long term (because we'll remove political zones).

Yes (provided that we remove political zones from legacy TIME as well).